extensions/windows-event-log/ConsumeWindowsEventLog.cpp

```diff
@@ -258,92 +389,65 @@ bool ConsumeWindowsEventLog::subscribe(const std::shared_ptr<core::ProcessContex
     logger_->log_info("inactiveDurationToReconnect: [%lld] ms", inactiveDurationToReconnect_);
   }
 
+  if (!pBookmark_) {
+    logger_->log_error("!pBookmark_");
+    return false;
+  }
+
+  auto channel = std::wstring(channel_.begin(), channel_.end());
+  auto query = std::wstring(query_.begin(), query_.end());
+
+  [&channel, &query, this]() {
```

Comment:
What is the purpose of this lambda if it is unnamed and called right away?

Reply:
I'll refactor.